### ISSUE TITLE
[consistent-type-imports] Allow type annotations in declaration blocks

### ISSUE THREAD

**Issue by glen-84** [typescript-eslint/typescript-eslint#3295]
I have code like this:

```ts
// types.d.ts
declare interface Window {
    Rollbar?: import("rollbar");
}
```

... which results in:

> `import()` type annotations are forbidden.

However, as far as I know, there is no alternative method of importing the type in a non-module declaration file. [It's not possible](https://stackoverflow.com/a/51114250/221528) to augment types in module-type declaration files, from what I understand.

So my request is to allow type imports in type annotations, if they are inside a `declare` block.

**Comment by bradzacher** [comment]
If you're augmenting window, can't you use `declare global`?

I don't have my computer handy but that is a namespace declaration then, meaning you should be able to import within it.

**Comment by glen-84** [comment]
That [requires](https://github.com/microsoft/TypeScript/issues/33128#issuecomment-526018445) converting `types.d.ts` to a module using something like `export {}`. Once that is done, other declarations stop working because it's no longer ambient (if I understand correctly).

**Comment by bradzacher** [comment]
I don't know if I understand everything correctly - but I don't think so?

We have many types files in this project that don't have `export {}`s nor `import ...`s at the top level such as https://github.com/typescript-eslint/typescript-eslint/blob/master/packages/eslint-plugin/typings/eslint-rules.d.ts

We just `include` the folder and it "just works":
https://github.com/typescript-eslint/typescript-eslint/blob/26d71b57fbff013b9c9434c96e2ba98c6c541259/packages/eslint-plugin/tsconfig.build.json#L11

**Comment by glen-84** [comment]
This is what happens if you add `declare global` to that file:

![image](https://user-images.githubusercontent.com/261509/115988598-c9054d00-a5ba-11eb-8105-66c885202b29.png)

Here is a basic example: https://github.com/glen-84/ts-es-issue-3295

Run `npm i` and see https://github.com/glen-84/ts-es-issue-3295/blob/master/types.d.ts.

**Comment by JoshuaKGoldberg** [comment]
Playground repro: https://typescript-eslint.io/play/#ts=5.1.6&fileType=.tsx&code=CYUwxgNghgTiAEBLAdgFxDAZlMCDqKwA9gO7wDeAUPPAEpEQQBGsA-AFzyoCeADiEUxIAtryIxUACgBEMBs1jSAlAG5KAX0pA&eslintrc=N4KABGBEBOCuA2BTAzpAXGUEKQAIBcBPABxQGNoBLY-AWhXkoDt8B6MgeyeUuX0Ra0ipWpQC2xDtHyoMAbUiJo0KZAA0mcNhwATXgEN48DgHcAKiUQBBJkw759%2BSl1lh8cRFogBfALpbvEG8gA&tsconfig=N4KABGBEDGD2C2AHAlgGwKYCcDyiAuysAdgM6QBcYoEEkJemy0eAcgK6qoDCAFutAGsylBm3TgwAXxCSgA&tokens=false

**Comment by bradzacher** [comment]
So the more I think about this - I think this is just an edge case where a disable comment is the correct thing.

It's valid TS code to `declare` an interface anywhere (regardless of if it's for decl merging or not).

So if we just allowed this then we'd create a hole in the rule where people could accidentally violate the codebase standard.

Its a rare enough case that you want to augment a global interface from within a non-module that I think I'm okay with it.

This issue has also garnered 0 reactions or comments from other community members in 3y - so it's not a common case for the broader community.
